### DISCUSSION TITLE
Grab basePath using nextjs?

### DISCUSSION THREAD

**Discussion by getnashty** [markdoc/markdoc#112] [Q&A] [answered]
I'm new to Markdoc, and Nextjs - but I've found both pretty easy to work with thus far. 

I'm trying to write some conditional documentation that changes based on the location.origin of the page.

First I tried creating a function that returned the window.location.origin - but window isn't available obviously, so that didn't work.

Is there a way to expose the current location origin, or nextjs basePath in a Markdoc document?

**Comment by mfix-stripe** [comment, marked as answer]
Hey @getnashty. We can't read the `location` itself on the Markdoc side since Markdoc isn't being run in the browser with the Next.js plugin. You can run Markdoc in the browser if you want though, that just isn't the idiomatic path. 

We do however surface the file path via `pageProps`, like:
```js
pageProps.markdoc?.file?.path
```

For your use case, I imagine you could just use custom tags that map to React components, and read from `location` in the React component before choosing which children to render.
